projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts

```diff
@@ -307,9 +281,13 @@ export class NgxIndexedDBService {
             const objectStore = transaction.objectStore(storeName);
 
             transaction.oncomplete = () => {
-              this.getByKey(storeName, key).subscribe((newValue) => {
-                resolve(newValue as T);
-              });
+              const request = objectStore.get(key) as IDBRequest<T>;
+              request.onsuccess = (event: Event) => {
+                resolve((event.target as IDBRequest<T>).result);
+              };
+              request.onerror = (event: Event) => {
+                reject(event);
+              };
```

Comment:
Why this change?

Reply:
Hi @assuncaocharles,
because this method also had the same bug reported by the issue #263. The getByKey method if called internally to another method returns promise resolved.